### PR TITLE
microblog with CRUD operation new PR

### PR DESCRIPTION
**Pull request by mayuriv-webonise**

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+    validates :title, presence: true
+    validates :content, presence: true
+    scope :published, -> { where(published: true) }
+    scope :unpublished, -> { where(published: false) }
+
+    def published?
+      published
```
Comment (kajalwebonise) [inline app/models/post.rb]: where we are using this method ?, if not then please remove it

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    end
+
+    def unpublished?
+      !published
```
Comment (kajalwebonise) [inline app/models/post.rb]: same here

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+    validates :title, presence: true
+    validates :content, presence: true
+    scope :published, -> { where(published: true) }
```
Comment (kajalwebonise) [inline app/models/post.rb]: same here

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    def published?
+      published
+    end
+
+    def unpublished?
+      !published
+    end
+
+    def publish
+      update(published: true)
```
Comment (kajalwebonise) [inline app/models/post.rb]: same here

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    end
+
+    def unpublish
+      update(published: false)
```
Comment (kajalwebonise) [inline app/models/post.rb]: same here

File: config/routes.rb [hunk 6]
```diff
@@ -1,6 +1,10 @@
...
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+  resources :posts do
+    member do
+      patch :publish
+      patch :unpublish
+    end
+  end
+end
```
Comment (kajalwebonise) [inline config/routes.rb:9]: add extra line

**Review by kajalwebonise — COMMENTED** [review]
please resolve the comment

File: app/views/posts/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<%= form_for @post do |f| %>
+  <div class="form-group">
+    <%= f.label :title %><br>
+    <%= f.text_field :title, class: 'form-control' %>
+  </div>
+  <div class="form-group">
+    <%= f.label :content %><br>
```
Comment (kajalwebonise) [inline app/views/posts/_form.html.erb:7]: in the db we are using title and body, and here we are using content, how come it is running ??

File: app/models/post.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
+class Post < ApplicationRecord
+    validates :title, presence: true
+    validates :content, presence: true
```
Comment (kajalwebonise) [inline app/models/post.rb:3]: in the db we are using title and body, and here we are using content, how come it is running ??

File: app/controllers/posts_controller.rb [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      redirect_to posts_path
+    end
+  
+    private
+    def set_post
+      @post = Post.find(params[:id])
+    end
+  
+    def post_params
+      params.require(:post).permit(:title, :content)
```
Comment (kajalwebonise) [inline app/controllers/posts_controller.rb:60]: in the db we are using title and body, and here we are using content, how come it is running ??

File: app/models/post.rb (hunk 8 above)
Comment (mayuriv-webonise) [reply app/models/post.rb:3]: Added missing file

File: app/controllers/posts_controller.rb (hunk 9 above)
Comment (mayuriv-webonise) [reply app/controllers/posts_controller.rb:60]: Added missing file

File: app/models/post.rb (hunk 1 above)
Comment (mayuriv-webonise) [reply app/models/post.rb]: remode

Comment (mayuriv-webonise) [reply app/models/post.rb]: removed

File: app/models/post.rb (hunk 2 above)
Comment (mayuriv-webonise) [reply app/models/post.rb]: removed

File: app/models/post.rb (hunk 3 above)
Comment (mayuriv-webonise) [reply app/models/post.rb]: removed

File: app/models/post.rb (hunk 4 above)
Comment (mayuriv-webonise) [reply app/models/post.rb]: removed

File: app/models/post.rb (hunk 5 above)
Comment (mayuriv-webonise) [reply app/models/post.rb]: removed

File: config/routes.rb (hunk 6 above)
Comment (mayuriv-webonise) [reply config/routes.rb:9]: done

File: app/views/posts/_form.html.erb (hunk 7 above)
Comment (mayuriv-webonise) [reply app/views/posts/_form.html.erb:7]: Added missing file

**Review by kajalwebonise — APPROVED** [review]

File: app/models/post.rb (hunk 8 above)
Comment (kajalwebonise) [reply app/models/post.rb:3]: ok

File: app/views/posts/_form.html.erb (hunk 7 above)
Comment (kajalwebonise) [reply app/views/posts/_form.html.erb:7]: ok

File: config/routes.rb (hunk 6 above)
Comment (kajalwebonise) [reply config/routes.rb:9]: ok

**Pull request merged**